### ISSUE TITLE
Properties not exported

### ISSUE THREAD

**Issue by Vpstr** [Dijji/FileMeta#50]
Hello!
I've recently installed FileMeta and tried to export some properties. I've set the handle for a file type (.mp4) and tried to export props to xml, but the resulting file turns out to be empty, although the file has plenty of unempty properties specified in the profile. The only thing in the xml is <Metafile/>.

**Comment by Dijji** [comment]
Have you set any properties on the file? Comments and Tags are good ones to try. They should appear in any exported meta data. At the moment, standard properties of files like MP4 will only be exported if you update their values.

Dijji

**Comment by Vpstr** [comment]
Yes, I have. I've put dozens of keywords into the file and my purpose for using this tool was to extract them. Some of the files I want to process also contain comments.

**Comment by Dijji** [comment]
Something special appears to be going on with MP4 files, and I suspect, also other video media files.
As far as I can tell, the usual registry entries that specify the property handler to use are being ignored by Explorer for MP4. When properties are edited, the values are always written to the main body of the file, and not the alternate stream, which is why the XML shows up empty, and the file size increases when properties are added.

This holds even if you use the Association Manager to add properties for display in Explorer which have nothing to do with the format (for instance, I’ve tried System.Actions.ActionName, and was able to edit and store the value without trouble).

Arguably, this behaviour is actually better than that normally provided by File Meta, as arbitrary metadata can be written to the file without risk of loss when it is, for instance, sent via email (I tried it).

There is thus no need for backup to XML. However, I do have another outstanding request to export properties held within the file to XML, so if you think this would be useful to you, let me know.

Dijji

**Comment by Vpstr** [comment]
So you mean that MP4 has native properties for keywords and comments? I doubt that , because these properties are shown in the tag view of explorer libraries.
I want to extract the properties anyway, because I need to convert these properties into some format that can be managed well by some program. Libraries are quite limited, for example I can't edit a keyword attached to several files for all these files at once. So I want to export them to a tagging program (I think of Hydrus) for this purpose. I presume I will find a possibility to push keywords into that program from XML file (though I'm not sure).

**Comment by Vpstr** [comment]
I can't edit them directly in the program, because they are already written into the files' properties and I don't want to extract them by hand. And I want to extract namely keywords and comments for each file separately i. e. I don't want them to mingle into a single file.

**Comment by Dijji** [comment]
Having thought about this, I decided that this was potentially too interesting a capability to bury somewhere deep in File Meta, so I implemented it as a new project which you can find here:

https://github.com/Dijji/ExtractFileProperties

Give it a try and let me know what you think. It probably makes sense to provide any feedback using Issues on the new project.

Dijji

**Comment by Dijji** [comment]
I have now investigated what is going on here. The results are described [here](../wiki/Windows-10-and-media-files). I include a description of what you can expect for now, and what File Meta is doing next.
